File: cpp/aircraft.cpp
#include "Aircraft.h"
#include <iostream>
#include <string>

using namespace std;

string Aircraft::getRoute() const{
    return route;
}


void Aircraft::displayInfo(std::vector<Transport> list) const  {
    Transport::displayInfo(list);
    cout << ", Route: " << route << "\n";

}

void Aircraft::displayAircraftInfo() const {
    cout << "Route: " << route << "\n";
}

File: java/aircraft.java
package cps2004.java;

public class aircraft extends transport {
    private String route;

    // Constructor
    public aircraft(String m, double n, double a, String x) {
        super(m, n, a);       
        this.route = x;
    }

    // Getter
    public String getRoute() {
        return route;
    }

    // Methods
    @Override
    public void displayInfo(transport transport) {
        super.displayInfo(transport);
        System.out.println(", Route: " + route);
    }

    public void displayAircraftInfo() {
        System.out.println("Route: " + route);
    }
}